src/main/java/org/jabref/gui/exporter/DeleteMetadataAction.java

```diff
@@ -0,0 +1,259 @@
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.BibEntryTypesManager;
+import org.jabref.preferences.FilePreferences;
+
+import static org.jabref.gui.actions.ActionHelper.needsDatabase;
+
+public class DeleteMetadataAction extends SimpleCommand {
+
+    private final StateManager stateManager;
+    private final DialogService dialogService;
+    private final TaskExecutor taskExecutor;
+    private final FilePreferences filePreferences;
+    private final XmpPreferences xmpPreferences;
+    private final EmbeddedBibFilePdfExporter embeddedBibExporter;
+
+    private OptionsDialog optionsDialog;
+
+    private BibDatabase database;
+    private Collection<BibEntry> entries;
+
+    private boolean shouldContinue = true;
+    private int skipped;
+    private int entriesChanged;
+    private int errors;
+
+    public DeleteMetadataAction(StateManager stateManager, BibDatabaseMode databaseMode, BibEntryTypesManager entryTypesManager, FieldWriterPreferences fieldWriterPreferences, DialogService dialogService, TaskExecutor taskExecutor, FilePreferences filePreferences, XmpPreferences xmpPreferences) {
+        this.stateManager = stateManager;
+        this.dialogService = dialogService;
+        this.taskExecutor = taskExecutor;
+        this.filePreferences = filePreferences;
+        this.xmpPreferences = xmpPreferences;
+        this.embeddedBibExporter = new EmbeddedBibFilePdfExporter(databaseMode, entryTypesManager, fieldWriterPreferences);
+
+        this.executable.bind(needsDatabase(stateManager));
+    }
+
+    @Override
+    public void execute() {
+        init();
+        BackgroundTask.wrap(this::deleteMetadata)
+                      .executeWith(taskExecutor);
+    }
+
+    public void init() {
+        if (stateManager.getActiveDatabase().isEmpty()) {
+            return;
+        }
+
+        database = stateManager.getActiveDatabase().get().getDatabase();
+        // Get entries and check if it makes sense to perform this operation
+        entries = stateManager.getSelectedEntries();
+
+        if (entries.isEmpty()) {
+
+            entries = database.getEntries();
+
+            if (entries.isEmpty()) {
+                dialogService.showErrorDialogAndWait(
+                        Localization.lang("Delete metadata from PDFs"),
+                        Localization.lang("This operation requires one or more entries to be selected."));
+                shouldContinue = false;
+                return;
+            } else {
+                boolean confirm = dialogService.showConfirmationDialogAndWait(
+                        Localization.lang("Delete metadata from PDFs"),
+                        Localization.lang("Delete metadata for all PDFs in current library?"));
+                if (confirm) {
+                    shouldContinue = false;
+                    return;
+                }
+            }
+        }
+
+        errors = entriesChanged = skipped = 0;
```

Comment:
Please split up into multiple lines - we do not have the joined assignements as code style.

Reply:
Done.